### PR TITLE
feat: support traces and spans

### PR DESCRIPTION
**Pull request by mdelapenya**
## What is this PR doing?
It uses the test suite struct representing the jUnit XML file, to send traces and spans using OpenTelemtry

For that:
- creates an outer span/transaction per execution
- stores the runtime as attributes, which will be added as labels to any span
- iterates the test suites in the struct, adding a span per test suite. Test suite attributes will be added
- for each test suite, iterates the tests elements, adding a span per test. Test attributes will be added

Finally, it adds docs and a docker-compose file to spin up an Elastic Stack to verify that traces/spans are there. The docs include 3 more comprehensive test results files to verify the system.

https://user-images.githubusercontent.com/951580/141785256-76dcfb7e-410c-46fc-a488-30f525ededab.mov

## How to test this?

```shell
# Run back-end for storing traces and spans (using Elastic Stack for this purpose)
docker-compose up -d
# Set the environment with the OTLP endpoint
eval $(cat test.env)
# Use the sample XML file and pass it to the binary
cat TEST-sample.xml | go run *.go
cat TEST-sample2.xml | go run *.go
cat TEST-sample3.xml | go run *.go
```

### CODE REVIEW

File: semconv.go [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 const (
+	// suite keys
 	FailedTestsCount  = "tests.failed"
 	ErrorTestsCount   = "tests.error"
 	PassedTestsCount  = "tests.passed"
 	SkippedTestsCount = "tests.skipped"
 	TestsDuration     = "tests.duration"
-	TestsName         = "tests.name"
-	TestsPackage      = "tests.package"
+	TestsSystemErr    = "tests.systemerr"
```
Comment (mdelapenya) [inline semconv.go:10]: We may need to contribute to OpenTelemtry's `semconv`, as it's very limited from test-results perspective.

@cyrille-leclerc thoughts on that?

**Review by kuisathaverat — APPROVED** [review]
LGTM

File: main.go [hunk 2]
```diff
@@ -28,6 +43,72 @@ func createIntCounter(meter metric.Meter, name string, description string) metri
...
+	meter := global.Meter("junit2otlp")
+
+	durationCounter := createIntCounter(meter, TestsDuration, "Duration of the tests")
+	errorCounter := createIntCounter(meter, ErrorTestsCount, "Total number of failed tests")
+	failedCounter := createIntCounter(meter, FailedTestsCount, "Total number of failed tests")
+	passedCounter := createIntCounter(meter, PassedTestsCount, "Total number of passed tests")
+	skippedCounter := createIntCounter(meter, SkippedTestsCount, "Total number of skipped tests")
+	testsCounter := createIntCounter(meter, TotalTestsCount, "Total number of executed tests")
+
+	ctx, outerSpan := tracer.Start(ctx, "junit2otlp", trace.WithAttributes(runtimeAttributes...))
```
Comment (v1v) [inline main.go:57]: NIT:

Can this be customised with something like `--name`?

Comment (v1v) [reply main.go:57]: Or `OTEL_SERVICE_NAME` overrides it?

**Review by v1v — APPROVED** [review]
LGTM

File: main.go (hunk 2 above)
Comment (mdelapenya) [reply main.go:57]: I think this is a good feature request. Let me open an issue for that.

Comment (mdelapenya) [reply main.go:57]: https://github.com/mdelapenya/junit2otlp/issues/8

**Pull request merged**